appium/webdriver/extensions/search_context.py

```diff
@@ -0,0 +1,117 @@
+
+
+class AndroidSearchContext(RootSearchContext):
+    """Define search context for Android"""
+
+    def find_element_by_android_data_matcher(self, name=None, args=None, className=None):
+        """Finds element by [onData](https://medium.com/androiddevelopers/adapterviews-and-espresso-f4172aa853cf) in Android
+        It works with [Espresso Driver](https://github.com/appium/appium-espresso-driver).
+
+        :Args:
+         - name - The name of a method to invoke. The method must return a Hamcrest [Matcher](http://hamcrest.org/JavaHamcrest/javadoc/1.3/org/hamcrest/Matcher.html)
+         - args - The args provided to the method
+         - className - The class name that the method is part of (defaults to `org.hamcrest.Matchers`).
+                       Can be fully qualified, or simple, and simple defaults to `androidx.test.espresso.matcher` package
+                       (e.g.: `class=CursorMatchers` fully qualified is `class=androidx.test.espresso.matcher.CursorMatchers`
+
+        :Returns:
+          An Element object
+
+        :Raises:
+        - TypeError - Raises a TypeError if the arguments are not validated for JSON format
+
+        :Usage:
+            driver.find_element_by_android_data_matcher(name='hasEntry', args=['title', 'Animation'])
+        """
+
+        return self.find_element(by=MobileBy.ANDROID_DATA_MATCHER, value=self._build_data_matcher(name=name, args=args, className=className))
+
+    def find_elements_by_android_data_matcher(self, name=None, args=None, className=None):
+        """Finds elements by [onData](https://medium.com/androiddevelopers/adapterviews-and-espresso-f4172aa853cf) in Android
+        It works with [Espresso Driver](https://github.com/appium/appium-espresso-driver).
+
+        :Args:
+         - name - The name of a method to invoke. The method must return a Hamcrest [Matcher](http://hamcrest.org/JavaHamcrest/javadoc/1.3/org/hamcrest/Matcher.html)
+         - args - The args provided to the method
+         - className - The class name that the method is part of (defaults to `org.hamcrest.Matchers`).
+                       Can be fully qualified, or simple, and simple defaults to `androidx.test.espresso.matcher` package
+                       (e.g.: `class=CursorMatchers` fully qualified is `class=androidx.test.espresso.matcher.CursorMatchers`
+
+        :Usage:
+            driver.find_elements_by_android_data_matcher(name='hasEntry', args=['title', 'Animation'])
+        """
+
+        return self.find_elements(by=MobileBy.ANDROID_DATA_MATCHER, value=self._build_data_matcher(name=name, args=args, className=className))
+
+    def _build_data_matcher(self, name=None, args=None, className=None):
+        result = {}
+
+        for key, value in {'name': name, 'args': args, 'class': className}.items():
+            if value is not None:
+                result[key] = value
+
+        return json.dumps(result)
+
+
+class IOSSearchContext(RootSearchContext):
```

Comment:
these empty classes are also not needed for now. We'll create them when there will be a need